ast/expressions.go

```diff
@@ -927,7 +927,10 @@ func (n *ValuesExpr) Accept(v Visitor) (Node, bool) {
 	if !ok {
 		return n, false
 	}
-	n.Column = node.(*ColumnNameExpr)
+	n.Column, ok = node.(*ColumnNameExpr)
+	if !ok {
```

Comment:
I prefer to put this check in `ColumnNameExpr` branch of `gbyResolver::Leave`, i.e, for a `ColumnNameExpr`, if the extracted result is not `ColumnNameExpr`, we should report error.

`Accept` acts as a general walker, it should not involve detailed visitor logic IMHO.